client-java/src/unit-tests/java/uk/gov/gchq/palisade/client/internal/request/RequestSerialisationTest.java

```diff
@@ -35,15 +33,15 @@ void setup() {
 
     @Test
     void testPalisadeResponseSerialisation() throws Exception {
-        var expected = createPalisadeResponse(b -> b.token("blah"));
+        var expected = PalisadeResponse.createPalisadeResponse(b -> b.token("blah"));
         var string = objectMapper.writeValueAsString(expected);
         var actual = objectMapper.readValue(string, PalisadeResponse.class);
         assertThat(actual).isEqualTo(expected);
```

Comment:
Much more useful for these simple objects to check they serialise to *exactly* the json we'd expect

Reply:
This test does not (and should not) care about the underlying JSON. All it cares about is whether the objects are the same after serialisation/deserialisation. I did for get the recursive equals though. Shameful of me.

Reply:
Fundamentally, the client still needs to send the correct json to the services - that is, all the correct key-value pairs as part of all the correct objects.
Part of the spec of the client is it can make REST and WS requests with specific JSON bodies such that the services will accept the requests.

It's not a unit-test, more like a contract-test, but this test should be done by the client at some point.

Reply:
Aren't we then just testing that Jackson is writing out the correct JSON? As long as the JSON is well formed, irrespective of ordering it should be accepted by the services. I think we do need a full contract test that loads up all 3 containers (palisade, frs, data) and runs a complete test. Maybe that's another ticket though.

Reply:
Yeah, we shouldn't care about ordering, but if the service expects `{userid: ..., resoruceid: ..., context: {...}}` etc... we should at least check that the key names are correct.
i.e. the client agrees with the message formats specified in the "Example JSON" sections of the palisade, filtered-resoure and data-service READMEs (mostly at the bottom of each):
https://github.com/gchq/Palisade-services/blob/develop/palisade-service/README.md
https://github.com/gchq/Palisade-services/blob/develop/filtered-resource-service/README.md
https://github.com/gchq/Palisade-services/blob/PAL-863-data-service-audit-message/data-service/README.md

Reply:
The result should be changing the member var names for the message classes in the client code here should cause a test failure

Reply:
Ahh, I see what you are getting at. I'll create a tests for these.

Reply:
If we had these objects create as a seperate `*-api.jar` from each service, all these classes and tests would be removed from the client. Maybe another discussion.

Reply:
Tests implemented that all the attributes of the generated json are correct. See the Serialisation tests in ~.resource, ~.download and ~.request packages in unit tests.